src/services/formatting/rules.ts

```diff
@@ -649,6 +662,28 @@ module ts.formatting {
             return context.TokensAreOnSameLine();
         }
 
+        static IsEndOfDecoratorContextOnSameLine(context: FormattingContext): boolean {
+            return context.TokensAreOnSameLine() &&
+                context.contextNode.decorators &&
+                Rules.NodeIsInDecoratorContext(context.currentTokenParent) &&
+                !Rules.NodeIsInDecoratorContext(context.nextTokenParent);
+        }
+
+        static NodeIsInDecoratorContext(node: Node): boolean {
+            if (node.parserContextFlags & ParserContextFlags.Decorator) {

```

Comment:
I thought it might be useful to be able to shortcut and save a few operations, but I can remove this.